scripts/compress_images.py

```diff
@@ -0,0 +1,255 @@
+
+                if new_size < original_size * self.tolerance:
+                    result_image.append({
+                        'path': file_path,
+                        'original_size': original_size,
+                        'new_size': new_size
+                    })
+            else:
+                logging.error(
+                    '[ERROR]: %s occurred on file %s',
+                        file_path,
+                        result.stderr
+                )
+
+        return result_image
+
+    def find_compressible_images(self) -> List[CompressedImageInfo]:
+        """Find images that can be compressed further.
+
+        Returns:
+            result_images: List[CompressedImageInfo]. List of all compressible
+            images over repository with attributes as path, original_size
+            and new_size.
+        """
+        result_images: List[CompressedImageInfo] = []
+
+        for file_path in self.input_path.glob('**/*.*'):
+            file_extension = file_path.suffix.lower()
+            if file_extension not in self.ALL_IMAGES_EXTENSIONS:
+                continue
+
+            get_tempdir = tempfile.gettempdir()
+            temp_compressed = (
+                pathlib.Path(get_tempdir) / f'compressed_{file_path.name}'
+            )
+            result = self.run_compression_logic(
+                file_path,
+                temp_compressed,
+            )
+            if result:
+                result_images.extend(result)
+
+        return result_images
+
+    def compress_images(
+            self, result_images: List[CompressedImageInfo]
+        ) -> None:
+        """Compress the identified images.
+
+        Args:
+            result_images: List[CompressedImageInfo]. List of images
+                to be compressed.
+        """
+        # Remove existing output directory and create a new one.
+        if os.path.exists(self.output_dir):
+            try:
+                shutil.rmtree(self.output_dir, ignore_errors=True)
+            except OSError as e:
+                logging.error(
+                    '[ERROR]: %s occurred on file %s',
+                    e, self.output_dir
+                )
+
+        os.makedirs(self.output_dir, exist_ok=True)
+
+        for image in result_images:
+            file_path = image['path']
+
+            # Maintain original directory structure.
+            rel_path = os.path.relpath(file_path)
+            output_file_path = pathlib.Path(self.output_dir) / rel_path
+
+            # Create necessary directories.
+            os.makedirs(os.path.dirname(output_file_path), exist_ok=True)
+            compress_result = self.run_compression_logic( # pylint: disable=unused-variable
+                file_path,
+                output_file_path,
+            )
+
+    def run(self) -> int:
+        """Main method to run the image compression process.
+
+        Returns: int. Exit code.
+        """
+        compressed_images = self.find_compressible_images()
```

Comment:
these images haven't been compressed yet, so the name is confusing. Maybe `image_compressibility_dicts`?